lib/3scale/client.rb

```diff
@@ -382,5 +390,24 @@ def build_error_response(body, klass = Response)
       response.error!(node.content.to_s.strip, node['code'].to_s.strip)
       response
     end
+
+    # Encode extensions header
+    def extensions_to_header(extensions)
+      {
+        '3scale-options'.freeze => extensions.map do |hk, hv|
```

Comment:
I don't have a strong preference.
I prefer to optimize for readability when the performance gain is not clear / not measured.